tests/test_end_to_end.py

```diff
@@ -510,3 +510,20 @@ def test_simple_table_headers(caplog, tmp_path):
             "headers to maintain Markdown spec"
         )
         assert any(expected_blurb in m for m in captured_messages)
+
+
+def test_link_mention_with_page(tmpdir):
```

Comment:
Very nice work creating a test! I would suggest that in this case we actually update the `test_all_blocks_page_to_markdown` to help keep the testing times quicker, since we already have a big test just for this page. With the unit tests, we definitely recommend small/quick tests, but with the end-to-end tests (since they're slow) sometimes we sacrifice having small atomic tests to make the test suite run faster.